### PR TITLE
[fixed] No materials on next map

### LINKED ISSUE #1344: No resupplies when starting a new match
## Description

One problem I see commonly in build 3900 is when I spawn in a new match as a builder I get no materials but the timer says 40 seconds remaining. I guess that the resupplies spawned but they were given to my teammate instead, because they overlap with my character.

Perhaps it's happening because my character had a full inventory in the previous match and the code in `CTF_GiveSpawnItems.as` SetMaterials() thinks my inventory is full, therefore merely places it at my position.

It should be checked if this specific issue was fixed by the recent resupply fix PR(s).

## Reproduction

Happens sometimes when a match ends and you are builder, then spawn into the next match with no materials.
Maybe happens always if your inventory was full in the previous match. 
Needs testing.

### PR DESCRIPTION
**Pull request by Gingerbeard5773**
## Status

- **READY**

## Description
Resolves #1344 & closes PR #1526

This PR decisively fixes the bug where players would spawn on next map without receiving their materials.

Why did the bug occur?
When going next map, players would be auto-balanced, and will change teams. Changing a player's team causes the player to request a new spawn. This kills their blob and spawns a new one. However, players receive their spawn mats before the second spawn happens, and they end up dropping their materials when they die.

TLDR: Players changing teams on next map causes them to drop their mats.

## Steps to Test or Reproduce

a) Go onto a dedicated server.
b) Spawn at least 3 bots with ``!bot``. You will probably need moderator/rcon access to use the chat command.
c) Keep going next map. If you are testing *without* the applied fix you will notice that you will receive no mats every time you switch teams by auto balance. Otherwise, with the applied fix you will receive mats as usual.

### CODE REVIEW

**Comment by AsuMagic** [comment]
.. That explains things :)

Why does the team swap logic kick in *after* that players have spawned? Could we reliably trigger it earlier without the game time check hack?

**Comment by Gingerbeard5773** [comment]
Its caused due to script priority in the gamemode config. DefaultBalanceTeams.as is triggered after CTF.as. I tried reordering scripts but it didn't work. The whole system is really sensitive and I didnt want to risk causing more bugs, which is why I ended up with a hack.

**Review by AsuMagic — APPROVED** [review]
:(
